lma-ai-stack/source/lambda_functions/share_meeting/index.py

```diff
@@ -0,0 +1,167 @@
+import os
+import io
+import boto3
+from boto3.dynamodb.conditions import Key, Attr
+from botocore.exceptions import ClientError
+import json
+import csv
+import logging
+import re
+from eventprocessor_utils import (
+    get_owner_from_jwt,
+)
+
+
+# grab environment variables
+LCA_CALL_EVENTS_TABLE = os.environ['LCA_CALL_EVENTS_TABLE']
+
+logger = logging.getLogger(__name__)
+ddb = boto3.resource('dynamodb')
+ddbTable = ddb.Table(LCA_CALL_EVENTS_TABLE)
+
+def update_meeting_permissions(callid, shardPK, shardSK, recipients):
+    pk = 'c#' + callid
+    new_recipients = list(set(email.strip() for email in recipients.split(',') if email.strip()))
+
+    try:
+        response = ddbTable.get_item(
+            Key={'PK': pk, 'SK': pk},
+            ProjectionExpression='SharedWith'
+        )
+        
+        current_recipients = response.get('Item', {}).get('SharedWith', [])
+
+        if set(new_recipients).issubset(set(current_recipients)):
+            print(f"Recipients already have access for {callid}")
+            return f"No update needed for CallId: {callid}"
+        
+        combined_recipients = list(set(current_recipients + new_recipients))
+
+        updated_count = update_recipients(pk, combined_recipients, None)
+        
+        pk = 'trs#' + callid
+        updated_count += update_recipients(pk, combined_recipients, None)
+
+        updated_count += update_recipients(shardPK, combined_recipients, shardSK)
+
+        print(f"Successfully updated {updated_count} items for CallId: {callid}")
+        return f"Updated {updated_count} items"
+
+    except ClientError as err:
+        logger.error("Error updating people can access %s: %s",
+                     err.response['Error']['Code'], err.response['Error']['Message'])
+        raise
+    else:
+        return response['Attributes']
+
+def update_recipients(pk, recipients, sk):
```

Comment:
We don't need to mutate this change through AppSync?  Wonder if directly updating DynamoDB would be problematics if we had Appsync Caching enabled, since the table update wouldn't update the cache (whereas mutate would)..

Reply:
Hmm, great point. Can definitely refactor to use the AppSync mechanism (and one of the reasons why I included the layer). Show stopper or punt to post-release?

Reply:
it's a two way door.. not a showstopper.. Log a story to update next release.